pkg/build/api/validation/validation.go

```diff
@@ -103,11 +103,18 @@ func validateBuildSpec(spec *buildapi.BuildSpec, fldPath *field.Path) field.Erro
 	allErrs := field.ErrorList{}
 	s := spec.Strategy
 
-	if s.CustomStrategy == nil && spec.Source.Git == nil && spec.Source.Binary == nil && spec.Source.Dockerfile == nil {
+	if s.CustomStrategy == nil && s.JenkinsPipelineStrategy == nil && spec.Source.Git == nil && spec.Source.Binary == nil && spec.Source.Dockerfile == nil {
```

Comment:
first off, it would never be a "reploy of jenkins service", it would be "update the jenkins job definition in the jenkins server" for both cases.

Now as for what component is responsible for detecting the change, when does it detect the change, and what does it do...well that's basically what i was trying to ask :)